lib/widgets/profile.dart

```diff
@@ -0,0 +1,261 @@
+  return switch (role) {
+    UserRole.owner => "Owner",
+    UserRole.administrator => "Administrator",
+    UserRole.moderator => "Moderator",
+    UserRole.member => "Member",
+    UserRole.guest => "Guest",
+    UserRole.unknown => "Unknown",
+  };
+}
+
+class _ProfileDataTable extends StatelessWidget {
+  const _ProfileDataTable({super.key, required this.user});
+
+  final User user;
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final profileData = user.profileData;
+    if (profileData == null) {
+      return const SizedBox.shrink();
+    }
+    final List<({String label, Widget widget})> entries = [];
+    for (final realmField in store.customProfileFields) {
+      final profileField = profileData[realmField.id];
+      if (profileField == null) continue;
+      final widget = _renderCustomProfileFieldValue(context, profileField.value, realmField);
+      if (widget == null) continue;
+      entries.add((label: realmField.name, widget: widget));
+    }
+
+    return DefaultTextStyle.merge(
+      style: const TextStyle(fontSize: 15, height: 1),
+      textAlign: TextAlign.left,
+      child: Column(
+        children: [
+          const SizedBox(height: 16),
+          Table(
+            border: TableBorder.all(style: BorderStyle.none),
+            columnWidths: const {
+              0: FixedColumnWidth(96 + 8),
+              1: FlexColumnWidth()},
+            defaultVerticalAlignment: TableCellVerticalAlignment.baseline,
+            textBaseline: TextBaseline.alphabetic,
+            children: entries.map((entry) => TableRow(
+              children: [
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(0, 4, 8, 4),
+                  child: Text(entry.label, style: const TextStyle(fontWeight: FontWeight.bold))),
+                entry.widget,
+              ])).toList()),
+        ]));
+  }
+
+  Widget? _renderCustomProfileFieldValue(BuildContext context, String value, CustomProfileField realmField) {
+    switch (realmField.type) {
+      case CustomProfileFieldType.link:
+        return _buildLinkWidget(context, value, value);
+      case CustomProfileFieldType.choice:
+        return _buildChoiceField(context, value, realmField);
+      case CustomProfileFieldType.externalAccount:
+        return _buildExternalAccountField(context, value, realmField);
+      case CustomProfileFieldType.user:
+        return _buildUserField(context, value, realmField);
+      case CustomProfileFieldType.date:
+        // TODO(server): The value's format is undocumented, but empirically
+        //   it's a date in ISO format, like 2000-01-01.
+        // That's readable as is, but:
+        // TODO format this date using user's locale.
+        return _buildTextWidget(value);
+      case CustomProfileFieldType.shortText:
+      case CustomProfileFieldType.longText:
+      case CustomProfileFieldType.pronouns:
+        return _buildTextWidget(value);
```

Comment:
```suggestion
      case CustomProfileFieldType.shortText:
      case CustomProfileFieldType.longText:
      case CustomProfileFieldType.pronouns:
        // The web client appears to treat `longText` identically to `shortText`;
        // `pronouns` is explicitly meant to display the same as `shortText`.
        return _buildTextWidget(value);
```

(Copying from the zulip-mobile implementation.) This comment is basically another piece of the reverse-engineering results that substitute here for API docs. Even though it doesn't involve any arbitrary names (like `choices`, `text`, `subtype`, etc., as seen for some of the other CustomProfileFieldType values), it's information about the semantics of these types which wouldn't be obvious a priori.